metals/src/main/scala/scala/meta/internal/metals/ReferenceProvider.scala

```diff
@@ -275,15 +284,40 @@ final class ReferenceProvider(
     buf.result()
   }
 
+  private def findRealRange(
+      range: s.Range,
+      text: String,
+      symbol: String
+  ): Option[s.Range] = {
+    val name = findName(range, text)
+    val isBackticked = name.charAt(0) == '`'
```

Comment:
you could extract `removeBackticks` function that would do the check and substring if needed.
it would be:
`val realName = removeBackticks(name)`
It could also be used in RenameProvider as removing backticks it the same there

Reply:
I tried it, but I still need to check for backticks and change the range, so I would check it double. This is on the critical path, so I'd rather leave it as is.

Reply:
ahh right isBackticked is needed later for another check, good as it is then :+1: